src/node_worker.cc

```diff
@@ -288,18 +251,9 @@ void Worker::Run() {
         env_->ProcessCliArgs(std::vector<std::string>{},
                              std::move(exec_argv_));
       }
-
+      this->env_ = env_.get();
```

Comment:
I think you need to either hold a lock for `mutex_` while setting this, or turn this and `stopped_` into a atomic fields and make sure everything is ordered correctly, otherwise you’ll have race conditions with `Worker::Exit()` (the same goes for resetting `stopped_` + `this->env_ = nullptr` above)

Reply:
done, covered under scopped mutex. I am still combing through the changes to see if there are still unattended writes.